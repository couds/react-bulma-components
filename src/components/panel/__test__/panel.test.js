import React from 'react';
import renderer from 'react-test-renderer';
import Panel from '..';

describe('Panel component', () => {
  it('Should Exist', () => {
    expect(Panel).toMatchSnapshot();
  });
  it('Should be a primary panel', () => {
    const component = renderer.create(
      <Panel color="primary">
        <Panel.Header>repositories</Panel.Header>
        <Panel.Block>
          <div>Control</div>
        </Panel.Block>
        <Panel.Tabs className="panel-tabs">
          <Panel.Tabs.Tab active>all</Panel.Tabs.Tab>
          <Panel.Tabs.Tab>public</Panel.Tabs.Tab>
          <Panel.Tabs.Tab>private</Panel.Tabs.Tab>
          <Panel.Tabs.Tab>sources</Panel.Tabs.Tab>
          <Panel.Tabs.Tab>forks</Panel.Tabs.Tab>
        </Panel.Tabs>
        <Panel.Block renderAs="a" active>
          <Panel.Icon>
            <i className="fa fa-bars" />
          </Panel.Icon>
          bulma
        </Panel.Block>
        <Panel.Block renderAs="label" className="panel-block">
          <input type="checkbox" />
          remember me
        </Panel.Block>
        <Panel.Block>
          <button type="button">reset all filters</button>
        </Panel.Block>
      </Panel>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should have box classname', () => {
    const component = renderer.create(
      <Panel>
        <Panel.Header>repositories</Panel.Header>
        <Panel.Block>
          <div>Control</div>
        </Panel.Block>
        <Panel.Tabs className="panel-tabs">
          <Panel.Tabs.Tab active>all</Panel.Tabs.Tab>
          <Panel.Tabs.Tab>public</Panel.Tabs.Tab>
          <Panel.Tabs.Tab>private</Panel.Tabs.Tab>
          <Panel.Tabs.Tab>sources</Panel.Tabs.Tab>
          <Panel.Tabs.Tab>forks</Panel.Tabs.Tab>
        </Panel.Tabs>
        <Panel.Block renderAs="a" active>
          <Panel.Icon>
            <i className="fa fa-bars" />
          </Panel.Icon>
          bulma
        </Panel.Block>
        <Panel.Block renderAs="label" className="panel-block">
          <input type="checkbox" />
          remember me
        </Panel.Block>
        <Panel.Block>
          <button type="button">reset all filters</button>
        </Panel.Block>
      </Panel>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
