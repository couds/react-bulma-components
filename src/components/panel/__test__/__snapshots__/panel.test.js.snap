// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel component Should Exist 1`] = `[Function]`;

exports[`Panel component Should be a primary panel 1`] = `
<nav
  className="panel is-primary"
>
  <div
    className="panel-heading"
  >
    repositories
  </div>
  <div
    className="panel-block"
  >
    <div>
      Control
    </div>
  </div>
  <div
    className="panel-tabs panel-tabs"
  >
    <a
      className="is-active"
    >
      all
    </a>
    <a>
      public
    </a>
    <a>
      private
    </a>
    <a>
      sources
    </a>
    <a>
      forks
    </a>
  </div>
  <a
    className="panel-block is-active"
  >
    <span
      className="panel-icon"
    >
      <i
        className="fa fa-bars"
      />
    </span>
    bulma
  </a>
  <label
    className="panel-block panel-block"
  >
    <input
      type="checkbox"
    />
    remember me
  </label>
  <div
    className="panel-block"
  >
    <button
      type="button"
    >
      reset all filters
    </button>
  </div>
</nav>
`;

exports[`Panel component Should have box classname 1`] = `
<nav
  className="panel"
>
  <div
    className="panel-heading"
  >
    repositories
  </div>
  <div
    className="panel-block"
  >
    <div>
      Control
    </div>
  </div>
  <div
    className="panel-tabs panel-tabs"
  >
    <a
      className="is-active"
    >
      all
    </a>
    <a>
      public
    </a>
    <a>
      private
    </a>
    <a>
      sources
    </a>
    <a>
      forks
    </a>
  </div>
  <a
    className="panel-block is-active"
  >
    <span
      className="panel-icon"
    >
      <i
        className="fa fa-bars"
      />
    </span>
    bulma
  </a>
  <label
    className="panel-block panel-block"
  >
    <input
      type="checkbox"
    />
    remember me
  </label>
  <div
    className="panel-block"
  >
    <button
      type="button"
    >
      reset all filters
    </button>
  </div>
</nav>
`;
